include/aws/http/request_response.h

```diff
@@ -356,6 +377,27 @@ struct aws_http1_chunk_options {
     void *user_data;
 };
 
+typedef aws_http_stream_write_complete_fn aws_h2_stream_write_data_complete_fn;
+
+/**
+ * Encoding options for manual H2 data frame writes
+ */
+struct aws_h2_data_write_options {
+    struct aws_input_stream *data;
+
+    /**
+     * Invoked when the data stream is no longer in use, whether or not it was successfully sent.
+     * Optional.
+     * See `aws_h2_stream_write_data_complete_fn`.
+     */
+    aws_h2_stream_write_data_complete_fn *on_complete;
+
```

Comment:
I can't recall and we discussed it, and actually thing a separate end_stream() function isn't necessarily a great idea since it will probably add extra work, an empty frame, etc, and doesn't map 1:1 with the protocol as well